DatadogSDKBridge/Classes/Implementation/DdRumImplementation.swift

```diff
@@ -95,17 +95,18 @@ internal class DdRumImplementation: DdRum {
     internal static let firstByteTimingKey = "firstByte"
     internal static let downloadTimingKey = "download"
 
-    let nativeRUM: NativeRUM
+    lazy var nativeRUM: NativeRUM = rumProvider()
+    private let rumProvider: () -> NativeRUM
 
     private typealias UserAction = (type: RUMUserActionType, name: String?)
     private var ongoingUserActions = [UserAction]()
 
-    internal init(_ nativeRUM: NativeRUM) {
-        self.nativeRUM = nativeRUM
+    internal init(_ rumProvider: @escaping () -> NativeRUM) {
+        self.rumProvider = rumProvider
```

Comment:
should we also set `Global.rum` (as discussed during the meeting today) here or somewhere else?